apps/api-extractor/src/collector/SourceMapper.ts

```diff
@@ -32,98 +49,59 @@ export class SourceMapper {
   private _originalFileInfoByPath: Map<string, IOriginalFileInfo> = new Map<string, IOriginalFileInfo>();
 
   /**
-   * If the `IExtractorMessageOptions` refers to a `.d.ts` file, look for a `.d.ts.map` and
-   * if possible update the coordinates to refer to the original `.ts` file.
+   * Given a `.d.ts` source file and a specific position within the file, return the corresponding
+   * `ISourceLocation`.
+   *
+   * @remarks
+   * If `useDtsLocation` is `false` (default), then we attempt to follow source maps in order to resolve
+   * the location to the original `.ts` file. If resolution isn't possible for some reason, we fall back
+   * to the `.d.ts` location.
+   *
+   * If `useDtsLocation` is `true`, then we don't bother following source maps, and the location refers
+   * to the `.d.ts` location.
    */
-  public updateExtractorMessageOptions(options: IExtractorMessageOptions): void {
-    if (!options.sourceFilePath) {
-      return;
-    }
+  public getSourceLocationFromFileAndPos(
+    sourceFile: ts.SourceFile,
+    pos: number,
+    useDtsLocation: boolean = false
+  ): ISourceLocation {
```

Comment:
We try to avoid designs where a function parameters are unclear. Instead of `getSourceLocationFromFileAndPos(file, 5, true)` it might be better to introduce an `options` parameter:

```ts
getSourceLocationFromFileAndPos(sourceFile, { position: 5, useDtsLocation: true })
```
or
```ts
getSourceLocationFromFileAndPos({ sourceFile, position: 5, useDtsLocation: true })
```

With such a design you could probably shorten the name to `getSourceLocation`.  If the operation has significant computational cost and is not cached, maybe use a verb like `calculateSourceLocation` or `lookupSourceLocation`.